### DISCUSSION TITLE
Help docker install (./K.sh: Syntax error: "do" unexpected)

### DISCUSSION THREAD

**Discussion by MrAdamo** [ctubio/Krypto-trading-bot#1041] [Q&A] [answered]
Hello!
I'm very new and inexperienced so please bear with me.
After reading and following the instructions I finally managed to install the bot in docker.
I use windows and never heard of docker before so that was a bit challenge for me to finally figure it out.
But after I finally managed to create the container, when I try to run it I get this error:
./K.sh: 3: ./K.sh: --exchange: not found
./K.sh: 9: ./K.sh: Syntax error: "do" unexpected
The exchange I use is coinbase pro so in the Dockerfile I put: ENV API_EXCHANGE COINBASE
I changed all the other API fields accordingly.

How do you choose the branch you want? I left the line untouched:
RUN git clone -b master https://github.com/ctubio/Krypto-trading-bot.git /K
I guess that is why my image file is 1.29GB

Also I didn't install boot2docker or docker-machine because in the docker docs webpage they say:
Machine was the only way to run Docker on Mac or Windows previous to Docker v1.12. Starting with the beta program and Docker v1.12, Docker Desktop for Mac and Docker Desktop for Windows are available as native apps and the better choice for this use case on newer desktops and laptops.

**Comment by ctubio** [comment]
morniinig'''
:koala: is like a year since i dont test this docker install; let me run it myself and see what is going on there.

meanwhile, i can't stop myself to say that docker install is not the recommended install method.

btw to change the branch from `master` to `ANOTHER` at:
`RUN git clone -b master https://github.com/ctubio/Krypto-trading-bot.git /K`
is:
`RUN git clone -b ANOTHER https://github.com/ctubio/Krypto-trading-bot.git /K`
(`-b` argument stands for branch, and its value is the name of the branch to be cloned; note the branch name does not need to be changed unless you develop your own branch and want to make use of it)

:koala: let me come back after retesting docker install'¡

**Reply by grhangelone** [reply]
Hi. You can use my repo, it's ready to run as long as you use x86.
https://hub.docker.com/r/grhangelone/ksh
I have the container running as service and it works perfectly. 
If we get another mayor version i will update it

**Reply by ctubio** [reply]
awesome lot of thanks'¡ (i will make use of it; let me not mark it as the final answer cos is not the very last version and is really a bug i introduced yesterday night)

:koala: i just need to charge my laptop with windows 10 (cos seems docker is not friendly on windows 7 anymore)

**Comment by ctubio** [comment, marked as answer]
@MrAdamo umf im sorry i always forget to update this line: https://github.com/ctubio/Krypto-trading-bot/blob/3880002b2430f5dd732f46db4f4da0b05d86b36f/Makefile#L236

was just fixed at https://github.com/ctubio/Krypto-trading-bot/commit/4275d8e5bde3a4cf1d67e0c2f7bf4acd3a0a2b5b, so the following errors will be gone:
```
./K.sh: 3: ./K.sh: --exchange: not found
./K.sh: 9: ./K.sh: Syntax error: "do" unexpected
```

**Reply by MrAdamo** [reply]
Thank you very much!
